src/loaders/compute_tools/microtrait/microtrait.py

```diff
@@ -107,6 +198,11 @@ def _run_microtrait(tool_safe_data_id: str, data_id: str, fna_file: Path, genome
 
     trait_counts_df.to_csv(os.path.join(genome_dir, loader_common_names.TRAIT_COUNTS_FILE), index=False)
 
+    heatmap_row, cells_meta, traits_meta = _process_trait_counts(trait_counts_df, data_id)
+    create_jsonl_files(genome_dir / MICROTRAIT_META, traits_meta)
+    create_jsonl_files(genome_dir / MICROTRAIT_CELLS, cells_meta)
+    create_jsonl_files(genome_dir / MICROTRAIT_DATA, heatmap_row)
```

Comment:
Could you post example (truncated) output?

Reply:
microtrait_cells.jsonl:
```
{"cell_id": "ede99274-cb61-45b2-9a76-6327570193b6", "values": "{}"}
{"cell_id": "43ef5193-9924-477a-ba7e-933ebb71440b", "values": "{\"exbD\": 116.4, \"exbB\": 191.9}"}
{"cell_id": "d9dc78fe-ea1a-4a59-ab2f-06be3c6a96bb", "values": "{}"}
{"cell_id": "a7a22542-0ed9-437b-8a7f-adf59478fb90", "values": "{\"TC.GNTP\": 429.9}"}
{"cell_id": "ec22e7ca-9473-41c2-8cfe-25b17275c449", "values": "{}"}
```

Reply:
microtrait_data.jsonl:

```
{"kbase_id": "69037_2_1", "cells": [{"cell_id": "ede99274-cb61-45b2-9a76-6327570193b6", "col_id": 1, "val": 0}, {"cell_id": "43ef5193-9924-477a-ba7e-933ebb71440b", "col_id": 2, "val": 2}, {"cell_id": "d9dc78fe-ea1a-4a59-ab2f-06be3c6a96bb", "col_id": 3, "val": 0}
...
{"cell_id": "b0a8ecb1-5167-4134-b1f5-c5301c1148d8", "col_id": 188, "val": 0}, {"cell_id": "80d6589e-98fd-4c12-8d6f-ccbaff1df37c", "col_id": 189, "val": 1}]}
```

Reply:
microtrait_meta.jsonl:

```
{"trait_index": 1, "trait_name": "Aromatic acid transport", "trait_description": "Resource Acquisition:Substrate uptake:aromatic acid transport", "trait_category": "Resource Acquisition", "trait_type": "count"}
{"trait_index": 2, "trait_name": "Biopolymer transport", "trait_description": "Resource Acquisition:Substrate uptake:biopolymer transport", "trait_category": "Resource Acquisition", "trait_type": "count"}
{"trait_index": 3, "trait_name": "Amino sugar transport", "trait_description": "Resource Acquisition:Substrate uptake:carbohydrate transport:amino sugar transport", "trait_category": "Resource Acquisition", "trait_type": "count"}
{"trait_index": 4, "trait_name": "Carbohydrate acid transport", "trait_description": "Resource Acquisition:Substrate uptake:carbohydrate transport:carbohydrate acid transport", "trait_category": "Resource Acquisition", "trait_type": "count"}
{"trait_index": 5, "trait_name": "Glycoside transport", "trait_description": "Resource Acquisition:Substrate uptake:carbohydrate transport:carbohydrate derivative transport:glycoside transport", "trait_category": "Resource Acquisition", "trait_type": "count"}

Reply:
For the cells why not unmarshal the gene values JSON?

Reply:
Oh and in data col_id is supposed to be a string

Reply:
👍 

new file samples:

microtrait_cells.jsonl:
```
{"cell_id": "059ed32d-7189-4569-b540-90c588498dd8", "values": {}}
{"cell_id": "8edfd2ee-0596-48b3-bc31-e235d63998c9", "values": {"exbB": 191.9, "exbD": 116.4}}
{"cell_id": "c2f73713-0ee9-4fab-9549-b7edcd507484", "values": {}}
{"cell_id": "808d8d3b-2286-4395-bfbe-edb331a73773", "values": {"TC.GNTP": 429.9}}
{"cell_id": "1f39f094-ed66-4ee3-b4ce-cfab33be72e5", "values": {}}
```

microtrait_data.jsonl:
```
{"kbase_id": "69037_2_1", "cells": [{"cell_id": "059ed32d-7189-4569-b540-90c588498dd8", "col_id": "1", "val": 0}, {"cell_id": "8edfd2ee-0596-48b3-bc31-e235d63998c9", "col_id": "2", "val": 2}, {"cell_id": "c2f73713-0ee9-4fab-9549-b7edcd507484", "col_id": "3", "val": 0}, {"cell_id": "808d8d3b-2286-4395-bfbe-edb331a73773", "col_id": "4", "val": 1},
```

microtrait_meta.jsonl:
```
{"col_id": "1", "name": "Aromatic acid transport", "description": "Resource Acquisition:Substrate uptake:aromatic acid transport", "category": "Resource Acquisition", "type": "count"}
{"col_id": "2", "name": "Biopolymer transport", "description": "Resource Acquisition:Substrate uptake:biopolymer transport", "category": "Resource Acquisition", "type": "count"}
{"col_id": "3", "name": "Amino sugar transport", "description": "Resource Acquisition:Substrate uptake:carbohydrate transport:amino sugar transport", "category": "Resource Acquisition", "type": "count"}
{"col_id": "4", "name": "Carbohydrate acid transport", "description": "Resource Acquisition:Substrate uptake:carbohydrate transport:carbohydrate acid transport", "category": "Resource Acquisition", "type": "count"}
{"col_id": "5", "name": "Glycoside transport", "description": "Resource Acquisition:Substrate uptake:carbohydrate transport:carbohydrate derivative transport:glycoside transport", "category": "Resource Acquisition", "type": "count"}
```

Reply:
hmmm `deconverted later`. where did we deconverted the detected_genes_score?

Reply:
They're dumped to json here:
https://github.com/kbase/collections/pull/419/files#diff-93fe7edc56038d1dac9d90291799622f2dd22ba247b451665c83c97e99c5da63R196-R197

and then deserialized here: https://github.com/kbase/collections/pull/419/files#diff-93fe7edc56038d1dac9d90291799622f2dd22ba247b451665c83c97e99c5da63R124

Why not just leave them as is, assuming a dict can be stored in pandas

Reply:
👍 removed json dump/loads. Generated jsonl files are the same as last commit. I wish I can remember why I did it at the first place. But it doesn't seem to break anything.

Reply:
:+1: